src/sentry/spans/buffer.py

```diff
@@ -280,18 +291,28 @@ def flush_segments(
                     if isinstance(val_data, dict):
                         val_data["__sentry_internal_old_segment_id"] = old_segment_id
                 val["segment_id"] = segment_span_id
-                val["is_segment"] = segment_span_id == val["span_id"]
+                is_segment = val["is_segment"] = segment_span_id == val["span_id"]
+
+                if old_segment_id:
+                    outcome = "same" if old_segment_id == segment_span_id else "different"
+                else:
+                    outcome = "null"
```

Comment:
I think in this case we would also want to know if there's a match or a mismatch, though it makes sense to me to separate this from valid segments.